src/framedata.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+/**
+ * The frame data header holds the classes that describe the members used for
+ * the actual logic of the frame tree tiling. So e.g. the 'selection' is a
+ * member of both of the following classes, whereas HSFrameLeaf::last_rect is
+ * not.
+ *
+ * Also note that the following classes only hold the plain members and does
+ * not set up class inheritance!
+ */
```

Comment:
I'm having a hard time understanding what this comment is trying to tell me. I'm aware that this is not a constructive review comment, but there's a circular dependency I'm struggling with...

Maybe there is more of an existing understanding expected of me (as in: the intended audience), so I'm not sure if my thoughts while reading this are helpful at all (and I'm not purposely trying to play dumb here):
 * What are "members used for the logic of the tiling"? Members as in functions?
 * Which "the 'selection'" is this about? Does that refer to the `selection_` member below?
 * What inheritance between which classe is *not* being set up here?

In other words, the crux might be that I have no idea what the question was that this is supposed to be the answer to.

Reply:
* I mean members as in 'member variables'. And the 'logic of the tiling' means: the following classes hold the member variables which are exposed to the user (via `dump`) and even explicitly in the man page in the section `TILING ALGORITHM`. In this section of the manpage, it is described what 'vertical' and 'horizontal' splittings mean, and thus this is part of the definition of the following classes. On the other hand, internal caching details like `HSFrameLeaf::last_rect` are just implementation details for caching certain things and this is thus in `HSFrameLeaf` and not in `FrameDataLeaf`
* The `selection` refers to both `FrameDataLeaf::selection` and to `FrameDataSplit<T>::selection_`.
* Note that HSFrameLeaf and HSFrameSplit inherit from a common base class HSFrame. But `FrameDataLeaf` and `FrameDataSplit<T>` do not have a common base class.

Reply:
The comment answers the question: why are some member variables part of `FrameDataSplit`/`FrameDataLeaf` and why are some other member variables part of `HSFrameSplit`/`HSFrameLeaf`.

Reply:
I've just clarified this comment.

Reply:
Thanks! I still feel like I'll have to dig a bit deeper some day to get the full picture, but at least this comment makes a lot more sense to me now. :+1: